### PR TITLE
AMICI dockerfile and deployment to dockerhub 

### PR DESCRIPTION
**Pull request by dweindl**
#909

### CODE REVIEW

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4978240&bid=16442446)

**Comment by FFroehlich** [comment]
Maybe I am misunderstanding the purpose of docker, but I don't see much use of anything but a Dockerfile. Are there really any realistic usecases where AMICI is used as standalone application?

**Comment by dweindl** [comment]
Given the recurring problems with installing AMICI, it might be helpful to some... Adding just the Dockerfile is also fine for me. Saves time :).

**Comment by FFroehlich** [comment]
> Given the recurring problems with installing AMICI, it might be helpful to some... Adding just the Dockerfile is also fine for me. Saves time :).

Will people that are not able to follow the current instruction guidelines be able to run docker images? I think this is better adressed by producing more informative error messages/tips when something fails during installation.

Dockerfiles is probably helpful for anybody building something on top of amici.

**Comment by dweindl** [comment]
> Will people that are not able to follow the current instruction guidelines be able to run docker images?

Not on me to judge.

> I think this is better adressed by producing more informative error messages/tips when something fails during installation.

That's complementary.
 
> Dockerfiles is probably helpful for anybody building something on top of amici.

Yes. And I know of several people who will have to run AMICI inside containers in the future...

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# AMICI with Docker
+
+## Create image
+
+In the AMICI base directory run:
+
+```bash
+git archive -o docker/amici.tar.gz --format=tar.gz HEAD
```
Comment (FFroehlich) [inline docker/README.md]: add this as CI test?

Comment (dweindl) [reply docker/README.md]: Planning to set up deployment to dockerhub. Would do that soonish, but separately.

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4978240&bid=16442446)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4978240&bid=16442446)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4978240&bid=16442446)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4978240&bid=16442446)

File: docker/README.md (hunk 1 above)
Comment (dweindl) [reply docker/README.md]: Done.

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4978240&bid=16442446)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4978240&bid=16442446)

**Comment by dweindl** [comment]
Sorry, few additional changes moved in here: The old PEtab test is now removed in favor of the benchmark collection tests on github. Before, travis runs took longer than allowed, due to installing PEtab dependencies.

**Comment by yannikschaelte** [comment]
alternatively: would a petab docker image be straightforward?

**Comment by dweindl** [comment]
> alternatively: would a petab docker image be straightforward?

The image itself, yes. Using it on travis seemed somewhat complicated last time I checked.

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4978240&bid=16442446)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4978240&bid=16442446)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4978240&bid=16442446)

**Comment by FFroehlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ICB-DCM/AMICI/pullRequest?prid=4978240&bid=16442446)

**Comment by wds15** [comment]
Cool to find docker support on this repo!

Since amici has a ton of dependencies this makes it a lot easier to use amici  as a code gen Blackbox which is useful when using the c++ interface I am interested in. Will try it out for sure.

**Pull request merged**